src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java

```diff
@@ -12,11 +16,35 @@ public MachineDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Machine save(Machine machine) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(machine);
+            transaction.commit();
+            return machine;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can`t add machine " + machine, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Machine> findByAgeOlderThan(int age) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            Query<Machine> machineOlderThan = session.createQuery("from Machine "
+                    + "where year < :age", Machine.class)
```

Comment:
```suggestion
                    + "where year < :dateFrom", Machine.class)
```
it will be more understandable

Reply:
Fixed